src/httpserver.cpp

```diff
@@ -91,7 +91,7 @@ class WorkQueue
     bool Enqueue(WorkItem* item)
     {
         LOCK(cs);
-        if (queue.size() >= maxDepth) {
+        if (!running || queue.size() >= maxDepth) {
```

Comment:
Well in that case the RPC error is also misleading. I'd like to avoid more calls to `ShutdownRequested`. Maybe just reword the error to something along "work queue full or shutdown in progress".